homeassistant/components/amazon_devices/coordinator.py

```diff
@@ -0,0 +1,84 @@
+"""Support for Amazon Devices."""
+
+from datetime import timedelta
+from typing import Any
+
+from aioamazondevices import AmazonDevice, AmazonEchoApi, exceptions
+from aioamazondevices.const import DEVICE_TYPE_TO_MODEL
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import _LOGGER, DOMAIN
+
+
+class AmazonDevicesCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """Base coordinator for Amazon Devices."""
+
+    config_entry: ConfigEntry
+    api: AmazonEchoApi
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        country: str,
+        username: str,
+        password: str,
+        login_data: dict[str, Any] | None = None,
+    ) -> None:
+        """Initialize the scanner."""
+
+        self.api = AmazonEchoApi(country, username, password, login_data)
+
+        self._login_username = username
+        self._login_country = country
+
+        super().__init__(
+            hass=hass,
+            logger=_LOGGER,
+            name=f"{DOMAIN}-{username}-coordinator",
+            update_interval=timedelta(seconds=30),
+        )
+        device_registry = dr.async_get(self.hass)
+        device_registry.async_get_or_create(
+            config_entry_id=self.config_entry.entry_id,
+            identifiers={(DOMAIN, self.config_entry.entry_id)},
+            name="Amazon Devices",
+            manufacturer="Amazon",
+        )
```

Comment:
I personally prefer this in `__init__.py`. can we use the account identifier as unique id?

Reply:
Changed the unique_id